experimental/oak_baremetal_channel/SPEC.MD

```diff
@@ -0,0 +1,290 @@
+
+Identifies the invocation the request initiates.
+
+##### method_id, u32, little endian
+
+Identifies the method that is being invoked.
+
+##### padding, 4 bytes
+
+Reserved to maintain 8 byte alignment. MUST be unset and ignored.
+
+##### body, variable length byte array
+
+Byte encoded method parameters.
+
+#### Response Message
+
+A byte encoded response message MUST consist of the following fields:
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                             length                            |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         invocation_id                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          status_code                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                            padding                            |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                         body bytes...                         +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ...
+```
+
+##### length u32, little endian
+
+Length of the entire message, including this field.
+
+##### invocation_id, u32, little endian
+
+MUST match the invocation_id of the relevant request message.
+
+##### status_code, u32, little endian
+
+response status as a
+[gRPC status code](https://grpc.github.io/grpc/core/md_doc_statuscodes.html).
+
+##### padding, 4 bytes
+
+Reserved to maintain 8 byte alignment. MUST be unset and ignored.
+
+##### body, variable length byte array
+
+For responses with an "Ok" status code the body MUST be the byte encoded method
+return value. Else the body MUST be a utf-8 encoded developer-facing error
+message.
+
+### Frame Layer
+
+Messages are sent and received as frames. Each message is fragmented into a set
+of one or more frames.
+
+#### Frame
+
+A byte encoded frame MUST consist of the following fields:
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                            padding                            |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|             length            |             flags             |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                         body bytes...                         +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ...
+```
```

Comment:
Is it possible to use hex for this? I think it would clarify where the byte boundaries are (but no worries if not easily possible). Something like

```
 0                               1                   
 0 1 2 3 4 5 6 7 8 9 A B C D E F 0 1 2 3 4 5 6 7 8 9 A B C D E F
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|                            padding                            |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|             length            |             flags             |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|                                                               |
+                         body bytes...                         +
|                                                               |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ...
```

Or maybe there are even better options that only count bytes instead of individual bits? Curious to see if the tool you used to generate it has any options for this.